File: New BackEnd/demo/src/main/java/com/example/demo/models/Worker.java
package com.example.demo.models;
import com.example.demo.embedded.Person;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "worker")
public class Worker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private String email;
    private String password;
    private String phone;
    private String profession;
    @Lob
    @Column( name = "image", length = 1000000000)
    private String image;
    @Embedded
    private Person generaleInfo;
    @ManyToMany
    private List<Activity> activities;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Company company;
}

File: photovoltaicV2/src/app/services/models/worker.ts
/* tslint:disable */
/* eslint-disable */
import {Activity} from '../models/activity';
import {Person} from '../models/person';

export interface Worker {
  activities: Array<Activity>;
  company: { id : string };
  email: string;
  generaleInfo: Person;
  id: number;
  image: string;
  password: string;
  phone: string;
  profession: string;
}
